### PR TITLE
Pythagorean triplet

### PR DESCRIPTION
**Pull request by softmoth**
Add the `pythagorean-triplet` example. The instructions and such are taken from the C track exercise. The _.meta/example.8th_ implementation is my own, based on the article mentioned at the top of the file.

Please let me know how to improve this so it can be included. I've also started a thread on the 8th forum for feedback: https://8th-dev.com/forum/index.php/topic,2738.0.html

I don't have an actual Exercism development environment set up, so it may need some tweaks in the config files, etc., to pass muster. I hope someone can help me to verify that it works OK.

### CODE REVIEW

**Comment by axtens** [comment]
I've reopened the PR so that we can progress it. Thank you very much for contributing. Given your skills in mentoring 8th, I expect you're not going to have much difficulty with adding examples. Having your input on the syllabus (see https://8th-dev.com/forum/index.php/topic,2726.0.html) would also be very much appreciated.

**Comment by axtens** [comment]
First comments: I am somewhat averse to the idea of having words defined in the *-tests.8th file itself. I'd rather extend the functionality of words in test-words.8th. My core reason is that the student is encouraged to comment out tests. If they see something other than a test in the tests file it might be confusing for them. 

That said, it's not a rule set in Roman self-healing concrete. Persuade me otherwise.

**Comment by axtens** [comment]
Also, if you haven't already joined the 8th forum, please do so. That way you get the ear of the developer plus input from people who've got wa-a-a-a-a-ay more experience than I have.

**Comment by softmoth** [comment]
Bruce, thanks for your thoughts. I'm on the 8th forum, but haven't been too active. I'll write up a request for feedback there to see what that turns up.

I've been thinking that with the `tests 7` at the top of the test file, the old advice of "comment out the tests you don't want" is already getting confusing. I have an idea to define `SKIP-REMAINING-TESTS` in `test-words.8th`, which will set the test counter to zero, write a message to the console, and `bye`. If we put that in every test file and change the instructions text, then it'll take care of the problem.

I'm also OK putting more into test-words.8th. I don't know if you're trying to keep that file the same for every exercise or not, let me know.

**Comment by axtens** [comment]
Whenever someone adds a new test word, I add it to the standard set and push the change to all exercises.

`SKIP-REMAINING-TESTS`, eh? That's persuasive.

**Comment by glennj** [comment]
The next steps for this PR:

1. Add the exercise to `config.json`. 
   * use the configlet tool to generate a UUID:
     ```sh
     bin/fetch-configlet
     bin/configlet uuid
     ```
2. use configlet to create the ancillary files:
   ```sh
   bin/configlet sync --exercise pythagorean-triplet --docs --update --yes
   bin/configlet sync --exercise pythagorean-triplet --metadata --update --yes
   bin/configlet sync --exercise pythagorean-triplet --tests include --update
   ```
3. populate `exercises/practice/pythagorean-triplet/.meta/config.json` similar to other exercises.

**Comment by glennj** [comment]
I also have some thoughts about skipping tests. Please have a look at 
https://github.com/exercism/8th/compare/exit-status...skip-tests

**Comment by softmoth** [comment]
I believe I have run all the `configlet` steps and all the metadata are up to date. So I'm done with this PR unless I've overlooked something. I'm happy to make any tweaks that would improve it.

Thanks!

Tim

File: exercises/practice/pythagorean-triplet/.meta/example.8th [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  rot 3 a:close
+;
+
+\ Pushes pythagorean triplet to the array at the top of the r-stack
+\ if its parts sum to n.
+: take-if-sum-equals \ n a --
+  swap
+  dup ' n:+ 0 a:reduce
+  rot n:=
+  if ensure-a<b dup . cr r@ swap a:push drop
```
Comment (glennj) [inline exercises/practice/pythagorean-triplet/.meta/example.8th:141]:
```suggestion
  if ensure-a<b r@ swap a:push drop
```
Need to print here?

**Review by glennj — APPROVED** [review]
With one small comment

**Comment by glennj** [comment]
Looks good to me. Bruce @axtens ?

**Comment by axtens** [comment]
If the both of you are good with it ... When I get home tonight I'll have a
look. (My timezone is UTC+8)

**Comment by axtens** [comment]
Does this need any tweaks now that I've merged the PR regarding the wholesale changes to the test framework?

**Comment by softmoth** [comment]
OK, this incorporates the main changes, including the latest `null? nip` fix.

**Review by glennj — APPROVED** [review]

**Pull request merged**